mars-rover-new/src/mars_rover_new/core.clj

```diff
@@ -1,7 +1,64 @@
+;; Mars Rover Exercise
+
 (ns mars-rover-new.core
+  (:require [clojure.string :as string])
   (:gen-class))
 
+;; todo: input validation, check invalid moves
+
+(declare add-plateau-size, add-rover-position, add-cmd-sequence)
+
+(def info-map {:plateau-size {:x nil
+                              :y nil}
+               :rover-position {:x nil
+                                :y nil
+                                :direction nil}
+               :cmd-sequence nil})
+
+(defn get-input
+  "Read one line of input stream. Trim and convert to upper-case."
+  []
+  (let [input (string/trim (read-line))]
+    (string/upper-case input)))
+
+(defn user-input
```

Comment:
[NON BLOCKER]
Can we think of better names for this function?

Reply:
Changed to process-user-input!